magenta/lib/melodies_lib.py

```diff
@@ -248,6 +254,10 @@ def _get_last_on_off_events(self):
         return (i, last_off)
     raise ValueError('No events in the stream')
 
+  def copy(self):
+    """Returns a deep copy of this MonphonicMelody object."""
```

Comment:
Yeah, if that's what we want, that's what we should call it.

Also, might not hurt to explicitly make a copy of the events list, just to be clear.